draft-ietf-ppm-dap.md

```diff
@@ -604,47 +724,54 @@ Clients upload reports by using an HTTP POST to `[leader]/upload`, where
 structured as follows:
 
 ~~~
+struct {
+    Time time;
+    Nonce nonce;
+    Extension extensions<0..2^16-1>;
+} ReportMetadata;
+
 struct {
   TaskID task_id;
-  Nonce nonce;
-  Extension extensions<0..2^16-1>;
+  ReportMetadata metadata;
   HpkeCiphertext encrypted_input_shares<1..2^32-1>;
 } Report;
 ~~~
 
-This message is called the client's report. It contains the following fields:
-
-* `task_id` is the task ID of the task for which the report is intended.
-* `nonce` is the report nonce generated by the client. This field is used by the
-  aggregators to ensure the report appears in at most one batch. (See
-  {{anti-replay}}.)
-* `extensions` is a list of extensions to be included in the Upload flow; see
-  {{upload-extensions}}.
-* `encrypted_input_shares` contains the encrypted input shares of each of the
-  aggregators. The order in which the encrypted input shares appear MUST match
-  the order of the task's `aggregator_endpoints` (i.e., the first share should
-  be the leader's, the second share should be for the first helper, and so on).
-
-To generate the report, the client begins by initializing the report nonce.
-Specifically, the client first sets `Nonce.rand` to 16 random bytes output from
-a cryptographically secure random number generator. It then sets `Nonce.time`
-to the number of seconds elapsed since the start of the UNIX epoch, rounded down to the nearest multiple of `min_batch_duration`.
-This truncation is done to ensure that the report's timestamp cannot be used to
-link the report back to the originating client.
-
-The client then finishes the report generation by sharding its measurement into
-a sequence of input shares as specified by the VDAF in use. To encrypt an input
-share, the client generates an {{HPKE}} ciphertext for the aggregator by running
+This message is called the Client's report. It consists of a "header" and the
```

Comment:
If we're going to refer to the header, why not actually introduce a header struct?

```
struct {
    TaskID task_id;
    Time time;
    Nonce nonce;
    Extension extensions<0..2^16-1>;
} ReportHeader;
```

Reply:
Also, this change seems unrelated to the query type change. Is it possible to apply it in a separate PR?

Reply:
That's a fine way to spell the header, but isn't the task ID also included in each report? In any case, we refer to this thing called a "header" but then don't actually define what it is. Let's either drop the word "header", or clarify that header == task ID + metadata. 

(And in the future, we need to stop landing unrelated changes in PRs like this!)

Reply:
The task ID is included in each `Report`, but is not sent in each `ReportShare` sent during the agg flow. OTOH the metadata is unique to each `ReportShare`, which is why I've split metadata and task ID.

Agreed on minimizing PRs. Like I said, it's an artifact of a previous version of this PR. Still, I think it's a useful change. I'm happy to pull it out into a separate PR.